pkgs/cli_config/lib/src/config.dart

```diff
@@ -200,6 +200,43 @@ class Config {
     );
   }
 
+  /// Constructs a config by parsing CLI arguments and loading the config file.
+  ///
+  /// The [arguments] must be commandline arguments.
+  ///
+  /// If provided, [environment] must be a map containing environment variables.
+  /// If not provided, [environment] defaults to [Platform.environment].
+  ///
+  /// If provided, [workingDirectory] is used to resolves paths inside
+  /// [environment].
+  /// If not provided, [workingDirectory] defaults to [Directory.current].
+  ///
+  /// This async constructor is intended to be used directly in CLI files.
```

Comment:
```suggestion
  /// This synchronous constructor is intended to be used directly in CLI files.
```
Maybe also add a reference to the `async` one?